modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/shibari_stand.dm

```diff
@@ -0,0 +1,269 @@
+/////////////////////
+//s-stand code here//
+/////////////////////
+
+/obj/structure/chair/shibari_stand
+	name = "shibari stand"
+	desc = "A stand for buckling people with ropes."
+	icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_structures/shibari_stand.dmi'
+	icon_state = "shibari_stand"
+	max_buckled_mobs = 1
+	max_integrity = 75
+	var/static/mutable_appearance/shibari_rope_overlay
+	var/static/mutable_appearance/shibari_rope_overlay_behind
+	var/static/mutable_appearance/shibari_shadow_overlay = mutable_appearance('modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_structures/shibari_stand.dmi', "shibari_shadow", OBJ_LAYER)
+	var/mob/living/carbon/human/current_mob = null
+	var/obj/item/stack/shibari_rope/ropee = null
+	var/current_color = "pink"
+
+// Default initialization
+/obj/structure/chair/shibari_stand/Initialize()
+	. = ..()
+	update_icon_state()
+	update_icon()
+
+/obj/structure/chair/shibari_stand/update_icon_state()
+	. = ..()
+	icon_state = "[initial(icon_state)]_[current_color]"
+
+/obj/structure/chair/shibari_stand/Destroy()
+	cut_overlay(shibari_shadow_overlay)
+	cut_overlay(shibari_rope_overlay)
+	. = ..()
+	if(current_mob)
+		if(current_mob.handcuffed)
+			current_mob.handcuffed.dropped(current_mob)
+		current_mob.set_handcuffed(null)
+		current_mob.update_abstract_handcuffed()
+	unbuckle_all_mobs(TRUE)
+
+/obj/structure/chair/shibari_stand/attack_hand(mob/living/user)
+	var/mob/living/attackedmobmeme = locate() in src.loc
+	if(!has_buckled_mobs())
+		if(attackedmobmeme?.can_buckle_to)
+			user_buckle_mob(attackedmobmeme, user, check_loc = TRUE)
+	else
+		var/mob/living/buckled_mob = buckled_mobs[1]
+		user_unbuckle_mob(buckled_mob, user)
+
+
+// Object cannot rotate
+/obj/structure/chair/shibari_stand/can_be_rotated(mob/user)
+	return FALSE
+// User cannot rotate the object
+/obj/structure/chair/shibari_stand/can_user_rotate(mob/user)
+	return FALSE
+// Another plug to disable rotation
+/obj/structure/chair/shibari_stand/attack_tk(mob/user)
+	return FALSE
+
+/obj/structure/chair/shibari_stand/user_unbuckle_mob(mob/living/buckled_mob, mob/living/user)
+	var/mob/living/buckledmeme = buckled_mob
```

Comment:
Make it a meaningful variable name that follows the snake_case naming convention.